Add subject prefixes to be removed

ikedas opened the pull request:
## Description

In this PR, I add variants in several languages of the prefixes like `RE:` / `Fwd:` that should be scraped off the subject. That is, those listed in [1][2][3] that I could actually find on the Internet, or some I have discovered independently.

In addition, several other improvements have been made: In some languages, a space is also placed before a colon [4]; a fullwidth colon is sometimes used after the prefix in Chinese ideographs; the space is not always placed after a colon. 

## Question

I think this PR could be applied to Znuny and OTOBO. Should I submit separate PRs for each?

## References

[1] Wikipedia: "[List of email subject abbreviations#Abbreviations in other languages](https://en.wikipedia.org/wiki/List_of_email_subject_abbreviations#Abbreviations_in_other_languages)".
[2] GNU Mailman 2.1: [Mailman/Handleres/CookHeaders.py](https://bazaar.launchpad.net/~mailman-coders/mailman/2.1/view/head:/Mailman/Handlers/CookHeaders.py#L424).
[3] Sympa 6.2: [Sympa/Regexps.pm](https://github.com/sympa-community/sympa/blob/6.2.70/src/lib/Sympa/Regexps.pm#L79-L81).
[4] [Bug \#893290](https://bugs.launchpad.net/mailman/+bug/893290).

ikedas committed:
Add subject prefixes to be removed

odyniec commented:
> In this PR, I add variants in several languages of the prefixes like `RE:` / `Fwd:` that should be scraped off the subject. That is, those listed in [1][2][3] that I could actually find on the Internet, or some I have discovered independently.

Thanks for this, this will surely be useful. However, there's an existing mechanism in place that assumes the different prefixes are stored in system configuration parameters rather that being hardcoded in `Kernel::System::Ticket`, and I think it would be better to extend these parameters to include the prefixes that you have collected. We'll try to do that for the next release -- let's keep this PR open until that happens.

> In addition, several other improvements have been made: In some languages, a space is also placed before a colon [4]; a fullwidth colon is sometimes used after the prefix in Chinese ideographs; the space is not always placed after a colon.

We'll make sure to take this into account (probably best to add a few tests for these cases).

> ## Question
> 
> I think this PR could be applied to Znuny and OTOBO. Should I submit separate PRs for each?

Yes, those projects are independent from us, so probably that's the best course of action.

ikedas committed:
Update unit test Ticket/TicketSubject

ikedas commented:
> > In this PR, I add variants in several languages of the prefixes like `RE:` / `Fwd:` that should be scraped off the subject. That is, those listed in [1][2][3] that I could actually find on the Internet, or some I have discovered independently.
> 
> Thanks for this, this will surely be useful. However, there's an existing mechanism in place that assumes the different prefixes are stored in system configuration parameters rather that being hardcoded in `Kernel::System::Ticket`, and I think it would be better to extend these parameters to include the prefixes that you have collected. We'll try to do that for the next release -- let's keep this PR open until that happens.

Thank you for being interested in this.  I look forward for the next replease!

> 
> > In addition, several other improvements have been made: In some languages, a space is also placed before a colon [4]; a fullwidth colon is sometimes used after the prefix in Chinese ideographs; the space is not always placed after a colon.
> 
> We'll make sure to take this into account (probably best to add a few tests for these cases).

I updated and added test cases.

> > ## Question
> > I think this PR could be applied to Znuny and OTOBO. Should I submit separate PRs for each?
> 
> Yes, those projects are independent from us, so probably that's the best course of action.

All right, I'll see how this PR ends up first and then decide what to do.
